test/lit/debug/full.wat

```diff
@@ -20,33 +20,26 @@
   ;; NRML-NEXT:   (drop
   ;; NRML-NEXT:    (i32.const 1)
   ;; NRML-NEXT:   )
-  ;; NRML-NEXT:   ;;@ src.cpp:3:4
```

Comment:
Looks like I misunderstood some of the source location stuff :)

Is it the case that a source location annotation applies to every following instruction in stack machine order until there's a new source location annotation? Do they continue applying beyond the beginnings and ends of control flow scopes?

Reply:
I think you may be referring to the issue fixed in #5906, see details there.

Reply:
Hmm, no, that PR seems to only affect the encoding of source map info, but I'm confused about how to properly parser source map info.

Reply:
I think the description there answers your question, though, unless I've misunderstood it. That is I think an annotation should *not* apply to following instructions, as the example there shows. However an annotation *should* apply to nested children iirc.

Reply:
This might be a bug in the current parser, but currently annotations do apply to the following instructions. For instance, try running the following module through `wasm-opt`:
```
(module
   (func $f
      ;;@ a:1:1
      (nop)
      (nop)
      (nop)
   )
)
````
I get the following ouput:
```
(module
 (type $0 (func))
 (func $f
  ;;@ a:1:1
  (nop)
  ;;@ a:1:1
  (nop)
  ;;@ a:1:1
  (nop)
 )
)
```

Reply:
Under the assumption that the old parser is working as intended, the new parser now matches that behavior as well. It's a little unclear to me whether that is the best behavior, though.